### PR TITLE
Bug 1683055: Bump deps, update generated crd, add update-codegen to Makefile

### PR DESCRIPTION
**Pull request by sallyom**
/cc @damemi
/cc @enj 

Follows pattern here: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/265
Will finally resolve bz 1683055, there's a nice description now in the CRD.

### CODE REVIEW

**Comment by damemi** [comment]
```
could not wait for build: the build service-ca-operator failed after 3m44s with reason DockerBuildFailed: Docker build strategy has failed. # github.com/openshift/service-ca-operator/vendor/github.com/openshift/library-go/pkg/crypto vendor/github.com/openshift/library-go/pkg/crypto/crypto.go:39:18: undefined: tls.VersionTLS13 # github.com/openshift/service-ca-operator/vendor/k8s.io/component-base/cli/flag vendor/k8s.io/component-base/cli/flag/ciphersuites_flag.go:80:18: undefined: tls.VersionTLS13 error: build error: running 'go build -o service-ca-operator ./cmd/service-ca-operator' failed with exit code 2
```

Looks like you've run into the same TLS error I did here: https://github.com/openshift/cluster-openshift-apiserver-operator/pull/210/commits/0ccf11ba5cb8e554907ecdb36003c6210338be9e#r302216519

**Comment by damemi** [comment]
For reference that Go build flag is needed because of [this change](https://github.com/openshift/library-go/blob/master/pkg/crypto/crypto.go#L34-L39) in library-go which adds TLS1.3, which will be enabled by default in Go 1.13.

**Comment by sallyom** [comment]
@damemi thanks for the info/help!

File: Dockerfile.rhel7 [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
+FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/service-ca-operator
 COPY . .
-RUN go build -o service-ca-operator ./cmd/service-ca-operator
+RUN GODEBUG=tls13=1 go build -o service-ca-operator ./cmd/service-ca-operator
```
Comment (enj) [inline Dockerfile.rhel7]: wat?

**Comment by enj** [comment]
/hold cancel

Google'd - debug var looks fine.  I believe you still need:

> Also openshift/release has to be updated to get new go versions IIRC.

/test all

**Comment by enj** [comment]
Diff seems larger than I would expect though, maybe you need a `glide cc` before doing the bump?

File: Dockerfile.rhel7 (hunk 1 above)
Comment (sallyom) [reply Dockerfile.rhel7]: see the note above from @damemi? with a bump in library-go, this is required or else when building the image you'll see 
```
STEP 4: RUN go build -o service-ca-operator ./cmd/service-ca-operator
# github.com/openshift/service-ca-operator/vendor/github.com/openshift/library-go/pkg/crypto
vendor/github.com/openshift/library-go/pkg/crypto/crypto.go:39:18: undefined: tls.VersionTLS13
# github.com/openshift/service-ca-operator/vendor/k8s.io/component-base/cli/flag
vendor/k8s.io/component-base/cli/flag/ciphersuites_flag.go:80:18: undefined: tls.VersionTLS13
error building at STEP "RUN go build -o service-ca-operator ./cmd/service-ca-operator": error while running runtime: exit status 2
```
weird, i agree.

**Comment by sallyom** [comment]
> Diff seems larger than I would expect though, maybe you need a `glide cc` before doing the bump?

Yes, I re-bumped and diff went from ~1300 to 235 files.

**Comment by enj** [comment]
/lgtm

Make sure to update release repo.

**Comment by enj** [comment]
Also, please fix this repo so that /approve works like the rest.

**Comment by sallyom** [comment]
> Also, please fix this repo so that /approve works like the rest.

https://github.com/openshift/release/pull/4524  working on it

**Comment by enj** [comment]
/close

Trying to kick CI.

**Pull request merged**